### ISSUE TITLE
Provide an OpenAPI schema

### ISSUE THREAD

**Issue by Eldemarkki** [Testaustime/testaustime-backend#105]
It would be nice to have an OpenAPI schema for the API, it would allow us to generate the types for the frontend automatically.

One possible library: https://github.com/juhaku/utoipa

Not sure what is the best way to distribute the schema, here are some ideas:
- Serve it from an endpoint
- Store it in the repository
  - Add checks in CI to ensure the schema is up-to-date
  - I would prefer this option so that I won't need to run the backend locally every time I want to get the latest schema

**Comment by Chicken** [comment]
Serving the OpenAPI schema from an API endpoint would make the most sense and would always be correct for the instance running. For local development you can just fetch the schema from production.

**Comment by Eldemarkki** [comment]
Yeah that's a good point, maybe it's better to serve it from an endpoint. I was thinking of cases where there the latest version of the backend has not yet been deployed to production but I guess those situations are pretty rare.
